File: services/dotreplit.js
class Service extends ServiceBase {
	constructor(...args) {
		super(...args)
		this.config = process.getDotreplitConfig()
	}
	async recv(cmd, _session) {
		if (cmd.dotReplitGetRequest) {
			return api.Command.create({dotReplitGetResponse: {dotReplit: this.config}})
		}
	}
}

const service = new Service(
	serviceInfo.id,
	serviceInfo.service,
	serviceInfo.name,
);
await service.start()

File: services/src/dotreplit.rs
pub struct DotReplit {}
use super::traits;

use anyhow::{format_err, Result};
use async_trait::async_trait;

#[async_trait]
impl traits::Service for DotReplit {
    async fn message(
        &mut self,
        _info: &super::types::ChannelInfo,
        message: goval::Command,
        _session: i32,
    ) -> Result<Option<goval::Command>> {
        let body = match message.body.clone() {
            None => return Err(format_err!("Expected command body")),
            Some(body) => body,
        };

        match body {
            goval::command::Body::DotReplitGetRequest(_) => {
                let mut dotreplit = goval::Command::default();
                let inner: goval::DotReplit = _info.dotreplit.read().await.clone().into();

                dotreplit.body = Some(goval::command::Body::DotReplitGetResponse(
                    goval::DotReplitGetResponse {
                        dot_replit: Some(inner),
                    },
                ));

                Ok(Some(dotreplit))
            }
            _ => Ok(None),
        }
    }
}
